src-v2/Judge/NTS.Judge/Adapters/Behinds/DashboardBehind.cs

```diff
@@ -63,6 +84,68 @@ async Task CreateOfficials(IEnumerable<Domain.Setup.Entities.Official> setupOffi
         }
     }
 
+    async Task CreateParticipations(Domain.Setup.Entities.Event setupEvent) 
+    {
+        foreach(var competition in setupEvent.Competitions)
+        {
+            var setupPhases = competition.Phases;
+            var competitionDistance = 0m;
+            var phases = new List<Phase>();
+            var categoryEntriesPairs = new Dictionary<AthleteCategory, List<RankingEntry>>
+            {
+                { AthleteCategory.Senior, new List<RankingEntry>() },
+                { AthleteCategory.Children, new List<RankingEntry>() },
+                { AthleteCategory.JuniorOrYoungAdult, new List<RankingEntry>() },
+                { AthleteCategory.Training, new List<RankingEntry>() }
+            };
+            foreach(var phase in setupPhases)
+            {
+                phases.Add(new Phase(phase.Loop!.Distance, phase.Recovery, phase.Rest, competition.Ruleset, setupPhases.Last() == phase, competition.CriRecovery));
```

Comment:
nested method invocations should be avoided as it's hard to tell which parameter belongs to which method and it leads to inferior debugging experience. Instead create the `Phase` as variable.